docs/en/custom-elements/introduction.md

```diff
@@ -0,0 +1,164 @@
+		onSignUp && onSignUp();
+	};
+
+	return (
+		<div key="root" classes={css.root}>
+			<div key="title" classes={css.title}>
+				{title || 'Subscribe Now'}
+			</div>
+			{signedUp ? (
+				<div key="confirmation" classes={css.thanks}>
+					Thank you for signing up!
+				</div>
+			) : (
+				<div key="signup" classes={css.signup}>
+					{signUpBonusDate && <div>Sign up by {signUpBonusDate.toLocaleDateString()}</div>}
+					<input type="text" />
+					<button onclick={signUp}>Sign Up</button>
+				</div>
+			)}
+		</div>
+	);
+});
+```
+
+To build a widget as a custom element, add a `build-widget` property to your `.dojorc` file and include the path to any widgets you want to expose as custom elements.
+
+> .dojorc
+
+```json
+{
+	"build-widget": {
+		"prefix": "ref",
+		"widgets": ["src/widgets/EmailSignup"]
+	}
+}
+```
+
+With this configuration in place, build the widgets using the Dojo CLI.
+
+```shell
+$ dojo build widget
+```
+
+The `output/dist/` directory now contains the built custom elements.
+
+| Filename                  | Description                                                                                                                                                                             |
+| ------------------------- | --------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| `ref-email-signup.js`     | This file includes all of the code required to hydrate your custom element. Loading this script in your HTML file is what makes your Dojo widget come alive as a custom element.        |
+| `ref-email-signup.js.map` | A JavaScript source map file for making debugging easier. These files are used only during debugging and do not need to be uploaded to a production environment.                        |
+| `ref-email-signup.css`    | The CSS used by the widget. Note that this includes the styles only for this one widget. If you are using a theme, you will need to separately include the theme CSS in your HTML page. |
+
+Note that that the `prefix` we added to `.dojorc` is used in the custom element tag name.
+
+## Using Custom Elements
+
+Using Custom Elements built with Dojo is as simple as including the generated files in your HTML. To use the email subscription widget, import the files.
+
+```html
+<!DOCTYPE html>
+<html lang="en-us" dir="ltr">
+<head>
+	<meta charset="utf-8">
+    <title>Using Custom Elements</title>
+    <link rel="stylesheet" href="./output/dist/email-signup-1.0.0.css">
+</head>
+<body>
+    <ref-email-signup id="emailSignup"></ref-email-signup>
+    <script async src="./output/dist/email-signup-1.0.0.js"></script>
+</body>
+</html>
+```
+
+If you have more than one custom element, include `link` and `script` tags for each element you are using.
+
+> On some browsers, you may have to include a custom elements shim. We recommend INSERT SHIM HERE
+
+### Attributes
```

Comment:
It feels like we should cover, `attributes`, `properties`, `events` and `children/slots` together in supplemental?